### ISSUE TITLE
keep field names in yaml

### ISSUE THREAD

**Issue by yaitskov** [ludat/conferer#63]
Now field name should be written in all lower case in yaml file.
It is not convenient to read because in Haskell composite field name words are outlined by Capital letter. Plus this is not intuitive at least for me.
If lowcasing whole field name is important I would convert field name to snake_case format.

**Comment by ludat** [comment]
That's right, the problem allowing `Key`s to have characters outside of lowercase ascii is that for some `Source`s there is no way of setting them.

for example right now there is an environment variables `Source`, and how would you fetch something that contains a underscore? since we use underscore as separators `warp.port` turns into `PREFIX_WARP_PORT` which means that casing and underscored kinda lose meaning.

In the latest version of `conferer-aeson` (which is the basis for `conferer-yaml`) I added some checks to verify that invalid key names are not present (otherwise they will be ignored).

And at last I think that having too long names for configuration is kind of an anti pattern: `databaseTlsCertificateExpirationDate` should probably be split into different records, DBConfig -> TLSConfig -> Certificate -> Expiration (a date), that would allow more reuse in the future.

All that said having `Key`s as the lower common denominator between `Source`s is not the perfect solution so I'll leave this issue open in case you (or anyone else can think of something better)
